lesson4/homeworklesson4.py

```diff
@@ -0,0 +1,76 @@
+def creating_a_string(first_string:str, second_string:str) -> str:
+    result_string = []
+    second_string = second_string[::-1]
+    if len(first_string) - len(second_string) > 0:
+        n = first_string[-(len(first_string) - len(second_string))::1]
+        for first_string, second_string in zip(first_string, second_string):
+            result_string.append(first_string + second_string)
+        result_string.append(n)
+    elif len(first_string) - len(second_string) < 0:
+        n = second_string[(len(first_string) - len(second_string))::1]
+        for first_string, second_string in zip(first_string, second_string):
+            result_string.append(first_string + second_string)
+        result_string.append(n)
+    else:
+        for first_string, second_string in zip(first_string, second_string):
+            result_string.append(first_string + second_string)
+    result_string = "".join(result_string)
+    return result_string
+
+def sort_list_of_int(test_list:list[int], condition:bool) -> list[int]:
```

Comment:
Я бы назвал второй параметр не `condition`, а `desc`, что означает descending order, т.е. обратный порядок. И сделал бы его по умолчанию равным False, то есть сигнатура функции бы выглядела так:

```suggestion
def bubble_sort(list_to_sort:list[int], desc:bool = False) -> list[int]:
```

Но тогда и нужны будут соответствующие изменения в последнем ифе